crates/types/src/rpc/transactions/broadcasted_invoke_transaction_v1.rs

```diff
@@ -0,0 +1,128 @@
+use crate::felt::{
+    Calldata, Felt, Nonce, TransactionHash, TransactionSignature, TransactionVersion,
+};
+use crate::rpc::transactions::invoke_transaction_v1::InvokeTransactionV1;
+use crate::rpc::transactions::BroadcastedTransactionCommon;
+
+#[derive(Debug, Clone, Eq, PartialEq, Deserialize, Serialize)]
+pub struct BroadcastedInvokeTransactionV1 {
+    #[serde(flatten)]
+    pub common: BroadcastedTransactionCommon,
+    pub sender_address: ContractAddress,
+    pub calldata: Calldata,
+}
+
+impl BroadcastedInvokeTransactionV1 {
+    pub fn new(
+        sender_address: ContractAddress,
+        max_fee: Fee,
+        signature: &TransactionSignature,
+        nonce: Nonce,
+        calldata: &Calldata,
+        version: TransactionVersion,
+    ) -> Self {
+        Self {
+            sender_address,
+            calldata: calldata.clone(),
+            common: BroadcastedTransactionCommon {
+                max_fee,
+                signature: signature.clone(),
+                nonce,
+                version,
+            },
+        }
+    }
+
+    pub fn create_sir_invoke_function(&self, chain_id: Felt) -> DevnetResult<SirInvokeFunction> {
+        Ok(SirInvokeFunction::new(
+            self.sender_address.into(),
+            EXECUTE_ENTRY_POINT_SELECTOR.clone(),
+            self.common.max_fee.0,
+            self.common.version.into(),
+            self.calldata.iter().map(|f| f.into()).collect(),
+            self.common.signature.iter().map(|f| f.into()).collect(),
+            chain_id.into(),
+            Some(self.common.nonce.into()),
+        )?)
+    }
+
+    pub fn create_invoke_transaction(
+        &self,
+        transaction_hash: TransactionHash,
+    ) -> InvokeTransactionV1 {
+        InvokeTransactionV1 {
+            transaction_hash,
+            max_fee: self.common.max_fee,
+            version: self.common.version,
+            signature: self.common.signature.clone(),
+            nonce: self.common.nonce,
+            sender_address: self.sender_address,
+            calldata: self.calldata.clone(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use serde::Deserialize;
+    use starknet_api::transaction::Fee;
+    use starknet_in_rust::definitions::block_context::StarknetChainId;
+
+    use crate::contract_address::ContractAddress;
+    use crate::felt::Felt;
+    use crate::rpc::transactions::broadcasted_invoke_transaction_v1::BroadcastedInvokeTransactionV1;
+    use crate::traits::ToHexString;
+
+    #[derive(Deserialize)]
+    struct FeederGatewayInvokeTransaction {
+        transaction_hash: Felt,
+        sender_address: Felt,
+        max_fee: Felt,
+        nonce: Felt,
+        calldata: Vec<Felt>,
+        version: Felt,
+    }
+
+    /// Get transaction from feeder gateway by hash and then using the same parameters compute the
+    /// transaction hash
+    #[test]
+    fn correct_transaction_hash_computation_compared_to_a_transaction_from_feeder_gateway() {
```

Comment:
Correct me if I'm wrong but the feeder gateway will be deprecated one day, right?

Reply:
Not sure